src/Http/TransferStatsHandler.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Http;
+
+use GuzzleHttp\TransferStats;
+use Magento\MagentoCloud\Filesystem\Driver\File;
+use Magento\MagentoCloud\Filesystem\FileList;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Callback for Guzzle TransferStats to log their time to first byte value.
+ */
+class TransferStatsHandler
+{
+    /** @var File */
+    private $file;
+
+    /** @var FileList */
+    private $fileList;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(File $file, FileList $fileList, LoggerInterface $logger)
+    {
+        $this->file = $file;
+        $this->fileList = $fileList;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Log stats from Guzzle request.
+     *
+     * @param TransferStats $stats
+     */
+    public function __invoke(TransferStats $stats)
+    {
+        $status = $stats->hasResponse() ? $stats->getResponse()->getStatusCode() : 'unknown';
+
+        if (300 < $status && $status < 400) {
```

Comment:
it seems more reasonable to pass only requests with `code == 200`

Reply:
We want to log responses that are `2xx`, `4xx`, or `5xx`. Really the need here was to skip any redirects since they would wind up getting logged twice